### PR TITLE
feat(dop): Supports automatic creation of pipelines

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Supports automatic creation of pipelines

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMjE4LDExOTBdLCJhc3NpZ25lZSI6WyIxMDAxMjYxIl19&id=310361&iterationID=1218&pId=301817&type=TASK)

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    Supports automatic creation of pipelines          |
| 🇨🇳 中文    |      支持自动创建流水线        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/dop/initialize.go [hunk 1]
```diff
@@ -743,25 +744,40 @@ func registerWebHook(bdl *bundle.Bundle) {
...
-		HookLocation: apistructs.HookLocation{
-			Org:         "-1",
-			Project:     "-1",
-			Application: "-1",
-		},
-	}
-	if err := bdl.CreateWebhook(ev); err != nil {
-		logrus.Warnf("failed to register guide_create event, %v", err)
-	}
+	//ev = apistructs.CreateHookRequest{
```
[the remark this answers is not in the document]
Comment (littlejiancc) [reply modules/dop/initialize.go:747]: done

File: modules/dop/providers/guide/provider.go [hunk 2]
```diff
@@ -38,12 +39,13 @@ type config struct {
...
-	DB       *gorm.DB           `autowired:"mysql-client"`
-	Register transport.Register `autowired:"service-register" required:"true"`
-	Trans    i18n.Translator    `translator:"project-pipeline" required:"true"`
+	Cfg             *config
+	Log             logs.Logger
+	bundle          *bundle.Bundle
+	DB              *gorm.DB           `autowired:"mysql-client"`
+	Register        transport.Register `autowired:"service-register" required:"true"`
+	Trans           i18n.Translator    `translator:"project-pipeline" required:"true"`
+	projectPipeline projectpipeline.Service
```
[the remark this answers is not in the document]
Comment (littlejiancc) [reply modules/dop/providers/guide/provider.go:48]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**